src/software/thunderscope/robot_communication.py

```diff
@@ -80,12 +85,40 @@ def __init__(
                 raise Exception("Could not find estop, make sure its plugged in")
 
     def __send_estop_state(self):
-        if not self.disable_estop:
-            while True:
-                self.current_proto_unix_io.send_proto(
-                    EstopState, EstopState(is_playing=self.estop_reader.isEstopPlay())
-                )
-                time.sleep(0.1)
+        """
+        Constant loop which sends the current estop status proto
+        Uses the keyboard estop value
+        Unless estop is plugged in, in which case the physical estop value overrides it
+        """
+        while True:
+            if not self.use_keyboard_estop:
+                self.estop_is_playing = self.estop_reader.isEstopPlay()
+
+            self.current_proto_unix_io.send_proto(
+                EstopState, EstopState(is_playing=self.estop_is_playing)
+            )
+            time.sleep(0.1)
```

Comment:
This might also be a good opportunity to test out sending `StopPrimitive` when estop is pressed. @itsarune thoughts?

Reply:
sure yeah we can add that in here